metpy/calc/tests/test_thermo.py

```diff
@@ -17,6 +17,17 @@
 from metpy.units import units
 
 
+def assertIsUnitNan(value, units):
```

Comment:
How about we lose the camel case? Also, can we put this in `metpy.testing`?

Also a name suggestion, just: `assert_nan`? Since we already have `assert_array_equal` (no units in name, but it handles it).